src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -994,7 +988,16 @@ private static ITypeReference GetArrayItemType(ArrayAccessSyntax syntax, IDiagno
                     }
 
                     // index was of the wrong type
-                    return ErrorType.Create(DiagnosticBuilder.ForPosition(syntax.IndexExpression).StringOrIntegerIndexerRequired(indexType));
+                    return InvalidAccessExpression(DiagnosticBuilder.ForPosition(syntax.IndexExpression).StringOrIntegerIndexerRequired(indexType), diagnostics, syntax.SafeAccessMarker is not null);
+
+                //case TupleType baseTuple when indexType is IntegerLiteralType integerLiteralIndex:
```

Comment:
Did you mean to include this?

Reply:
This caused some baseline changes, so I meant to move it to another branch for a future PR.